File: UniqueEmailAddress.py
class Solution:
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        s = set()
        for i in range(len(emails)):
            print(emails[i])
            x = emails[i]
            print(x)
            if('+' in x):
                index = x.find('+')
                local = x[:index].replace(".","")
                index = x.find('@')
                domain = x[index:]
                email_ = local + domain
                s.add(email_)
            else:
                index = x.find('@')
                local = x[:index].replace(".","")
                domain = x[index:]
                email_ = local + domain
                s.add(email_)
        return len(s)
            

File: UniqueEmailAddress.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        set<string> s;
        for(int i = 0; i<emails.size(); i++){
            string email = emails[i];
            string x ="";
            bool domain = false;
            for(int j = 0; j<email.length();j++){
                if(email[j] == '+' && !domain){
                    while(email[j]!= '@'){
                        j++;
                        domain = true;
                    }
                }
                if(email[j] == '@'){
                    domain= true;
                }
                if(email[j] == '.' && !domain){
                    continue;
                }
                if(email[j] == '.' && domain){
                    x = x+ '.';
                    continue;
                }
                x = x+ email[j];
            }
            s.insert(x);
        }
        return s.size();
    }
};